src/System.Net.Utilities/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -2,33 +2,259 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+using System.Text;
 using System.Threading;
 
 namespace System.Net.NetworkInformation
 {
     public partial class Ping
     {
+        private const int IcmpHeaderLengthInBytes = 8;
+        private const int IpHeaderLengthInBytes = 20;
+
+        private static readonly bool s_canUseRawSockets = CheckRawSocketPermissions();
+
+        private static readonly string s_ipv4PingFile = "ping";
+        private static readonly string s_ipv6PingFile = "ping6";
+
+        private static bool CheckRawSocketPermissions()
```

Comment:
Please follow the Common\test\System\Net\Capability.cs partial class model instead of adding this here.